de.metas.async/src/main/java/de/metas/async/api/impl/AbstractQueueDAO.java

```diff
@@ -340,6 +343,24 @@ public void save(I_C_Queue_Element element)
 		return result;
 	}
 
+	@Override
+	public final List<Integer> retrieveAllItemIDs(final I_C_Queue_WorkPackage workPackage)
+	{
+		final List<I_C_Queue_Element> queueElements = retrieveQueueElements(workPackage, false);
+		return queueElements.stream()
+				.peek(this::assertTableId)
+				.map(I_C_Queue_Element::getRecord_ID)
+				.collect(ImmutableList.toImmutableList());
+	}
+
+	private void assertTableId(@NonNull final I_C_Queue_Element element)
```

Comment:
why this method? AD_Table_ID is mandatory.
More, if it would not, i would call it assertTableIdIsNotNull.